1-mandatory/1-writers.js

```diff
@@ -62,7 +64,11 @@ Exercise 2:
 */
 
 function logDeadWritersInTheirForties() {
-  // write your code here
+  const deadOlderWriters = writers.filter((eachWriter) => eachWriter.age > 39 && eachWriter.age < 50 && eachWriter.alive === false);
+  const writerName = deadOlderWriters.map((eachWriter) => eachWriter.firstName);
+  const writerSurname = deadOlderWriters.map((eachWriter) => eachWriter.lastName);
+  const writerAge = deadOlderWriters.map((eachWriter) => eachWriter.age);
+  console.log(`Writer ${writerName} ${writerSurname} died at ${writerAge} years old.`)
 }
```

Comment:
Well done of  using map, but I think this way is a bit complicated if you are using a loop and if state it would be much easier something like this `function logDeadWritersInTheirForties() {
  for (let i = 0; i < writers.length; i++) {
    if (
      writers[i].age >= 40 &&
      writers[i].age < 50 &&
      writers[i].alive === false
    ) {
      console.log(
        `Writer ${writers[i].firstName} ${writers[i].lastName} died at ${writers[i].age} years old.`
      );
    }
  }
}
`